tensorflow_addons/optimizers/rectified_adam.py

```diff
@@ -175,7 +175,7 @@ def _resource_apply_dense(self, grad, var):
         beta_2_power = tf.pow(beta_2_t, local_step)
 
         if self._initial_total_steps > 0:
-            total_steps = self._get_hyper("total_steps", var_dtype)
+            total_steps = int(self._get_hyper("total_steps", var_dtype))
```

Comment:
Why it needs to be casted to int? I think following computation all in `var_dtype`.